src/mcp/shared/session.py

```diff
@@ -283,6 +312,13 @@ async def send_notification(
         self,
         notification: SendNotificationT,
         related_request_id: RequestId | None = None,
+    ) -> None:
+        await self._send_notification_internal(notification, related_request_id)
+
+    async def _send_notification_internal(
```

Comment:
This code is ugly python type checking seems to make it necessary - alternative might be attempt some sort of sorcery with involving targetted disabling type checking